spring-xd-test/src/main/java/org/springframework/xd/test/kafka/KafkaTestSupport.java

```diff
@@ -49,81 +62,99 @@
-		return kafkaServer;
-	}
-
 	public String getBrokerAddress() {
-		return getKafkaServer().config().hostName() + ":" + getKafkaServer().config().port();
+		if (embedded) {
+			return kafkaServer.config().hostName() + ":" + kafkaServer.config().port();
+		}
+		else {
+			return DEFAULT_KAFKA_CONNECT;
+		}
 	}
 
 	@Override
 	protected void obtainResource() throws Exception {
-		try {
-			log.debug("Starting Zookeeper");
-			zookeeper = new EmbeddedZookeeper(TestZKUtils.zookeeperConnect());
-			log.debug("Started Zookeeper at " + zookeeper.getConnectString());
+		if (embedded) {
+			try {
+				log.debug("Starting Zookeeper");
+				zookeeper = new EmbeddedZookeeper(TestZKUtils.zookeeperConnect());
+				log.debug("Started Zookeeper at " + zookeeper.getConnectString());
+			}
+			catch (Exception e) {
+				throw new RuntimeException("Issues creating the ZK server", e);
+			}
+			try {
+				int zkConnectionTimeout = 6000;
+				int zkSessionTimeout = 6000;
+				zkClient = new ZkClient(getZkConnectString(), zkSessionTimeout, zkConnectionTimeout, ZKStringSerializer$.MODULE$);
+			}
+			catch (Exception e) {
+				zookeeper.shutdown();
+				throw new RuntimeException("Issues creating the ZK client", e);
+			}
+			try {
+				log.debug("Creating Kafka server");
+				Properties brokerConfigProperties = brokerConfig;
+				kafkaServer = TestUtils.createServer(new KafkaConfig(brokerConfigProperties), SystemTime$.MODULE$);
+				log.debug("Created Kafka server at " + kafkaServer.config().hostName() + ":" + kafkaServer.config().port());
+			}
+			catch (Exception e) {
+				zookeeper.shutdown();
+				zkClient.close();
+				throw new RuntimeException("Issues creating the Kafka server", e);
+			}
 		}
-		catch (Exception e) {
-			throw new RuntimeException("Issues creating the ZK server", e);
-		}
-		try {
-			int zkConnectionTimeout = 6000;
-			int zkSessionTimeout = 6000;
-			zkClient = new ZkClient(getZkConnectString(), zkSessionTimeout, zkConnectionTimeout, ZKStringSerializer$.MODULE$);
-		}
-		catch (Exception e) {
-			zookeeper.shutdown();
-			throw new RuntimeException("Issues creating the ZK client", e);
-		}
-		try {
-			log.debug("Creating Kafka server");
-			Properties brokerConfigProperties = brokerConfig;
-			kafkaServer = TestUtils.createServer(new KafkaConfig(brokerConfigProperties), SystemTime$.MODULE$);
-			log.debug("Created Kafka server at " + kafkaServer.config().hostName() + ":" + kafkaServer.config().port());
-		}
-		catch (Exception e) {
-			zookeeper.shutdown();
-			zkClient.close();
-			throw new RuntimeException("Issues creating the Kafka server", e);
+		else {
+			try {
+				this.zkClient = new ZkClient(DEFAULT_ZOOKEEPER_CONNECT, 5000, 5000, ZKStringSerializer$.MODULE$);
+				if (ZkUtils.getAllBrokersInCluster(zkClient).size() == 0) {
+					throw new RuntimeException("Kafka server not available on localhost");
+				}
+			}
+			catch (Exception e) {
+				throw new RuntimeException("Issues connecting to ZK server");
+			}
```

Comment:
See comment above about wrapping exception (in fact this one doesn't even wrap the original exception).